GenderPayGap.UnitTests/GenderPayGap.WebUI.Tests/CodeQualityTests/DoNotUseRedirectWithReturnUrls.cs

```diff
@@ -45,19 +46,22 @@ public void DoNotAllowRedirectWithReturnUrlInCode()
                     continue;
                 }
 
-                string fileText = File.ReadAllText(filePath);
-                // There are a couple of actions on controllers Redirect() that wrap a RedirectToAction,
-                // we want to exclude these from the check as we are only concerned about redirecting with a url parameter
-                if (fileText.Contains(" Redirect(") && !fileText.Contains(" Redirect()"))
+                // We want to pick up all instances of Redirect(url) or new RedirectResult(url) that do not have the disable comment on the previous line.
+                // Ignores those without parameters so we don't pick up any methods defined as Redirect() for example
+                var lines = File.ReadAllLines(filePath);
+                for (var i = 0; i<lines.Length; i++)
```

Comment:
I'm a bit worried about the speed/efficiency impact of this - C# will have all sorts of weird internal optimisations for the regex-only solution, but manually looping through the lines will lose it.

Can you think of a regex that matches only the right cases? Also, how long does the test take to run on your machine? (It's not production code so I'm not too bothered if a regex solution isn't obvious)

Reply:
Alternatively, how about 
* First pass: does it match the redirect regex at all?
* If it does, do a second pass - go through all the lines in just that one file and do this check?